tests/end_to_end_test/restore_test.go

```diff
@@ -150,6 +155,33 @@ func TestSnapshotRestore(t *testing.T) {
 	// Restore last snapshot using the snapshot ID
 	e.RunAndExpectSuccess(t, "snapshot", "restore", snapID, restoreDir)
 
+	restoreArchiveDir := makeScratchDir(t)
+	zipFile := filepath.Join(restoreArchiveDir, "output.zip")
+	e.RunAndExpectSuccess(t, "snapshot", "restore", snapID, zipFile)
+	verifyValidZipFile(t, zipFile)
+
+	tarFile := filepath.Join(restoreArchiveDir, "output.tar")
+	e.RunAndExpectSuccess(t, "snapshot", "restore", snapID, tarFile)
+	verifyValidTarFile(t, tarFile)
+
+	targzFile := filepath.Join(restoreArchiveDir, "output.tar.gz")
+	e.RunAndExpectSuccess(t, "snapshot", "restore", snapID, targzFile)
+	verifyValidTarGzipFile(t, targzFile)
+
+	tgzFile := filepath.Join(restoreArchiveDir, "output.tgz")
+	e.RunAndExpectSuccess(t, "snapshot", "restore", snapID, tgzFile)
+	verifyValidTarGzipFile(t, tgzFile)
+
+	// create a directory whose name ends with '.zip' and override mode to force treating it as directory.
+	zipDir := filepath.Join(restoreArchiveDir, "outputdir.zip")
+	e.RunAndExpectSuccess(t, "snapshot", "restore", snapID, zipDir, "--mode=local")
+
+	// verify we got a directory
+	st, err := os.Stat(zipDir)
+	if err != nil || !st.IsDir() {
+		t.Errorf("unexpected stat() results on output.zip directory %v %v", st, err)
+	}
+
```

Comment:
These should be done after `compareDirs` below, since `compareDirs` checks the restore to directory from above.

Reply:
done